core/request_handler.go

```diff
@@ -110,7 +110,12 @@ func (c *Connection) processRequest(ch *Channel, req *vppRequest) error {
 	}
 
 	// send the request to VPP
-	t := time.Now()
+	c.trace.record(&api.Record{
```

Comment:
If the `SendMsg` fails, how will user know? What is the reasoning behind putting trace record of sending message before we actually send the message? It will pollute the records with message requests that could have actually failed.

Reply:
It's not the tracer's purpose to inform the user about failed messages but about messages being sent. My take on this is that failed messages should not be ignored.

Reply:
@VladoLavor While I understand your perspective on the importance of capturing failed messages, it's worth considering the purpose of trace events. Tracing is primarily intended to **record the sequence of events** and provide visibility into the flow of messages.

By recording a trace event **before actually sending the message**, we risk including messages that were **never sent in the first place**. This could potentially clutter the trace records and make it difficult to distinguish between successful and failed messages.

Instead, it might be more effective to focus on capturing trace events **after the message has been sent**. This way, we ensure that the recorded events accurately reflect the messages that were actually sent, providing a clearer understanding of the system's behavior.